pdm/cli/commands/run.py

```diff
@@ -21,9 +21,14 @@
 from pdm.utils import is_path_relative_to
 
 
+class EnvFileOptions(TypedDict, total=True):
+    path: str
+    override: bool | None
```

Comment:
I recommend:

```python
class EnvFileOverride(TypedDict, total=True):
    override: str
```
This means when user uses a table, `env_file.override = ".env"` , then the env will override. If he doesn't want to override, he will use a plain text `env_file = ".env"`

Reply:
I can't get this to work (I might be missing something), the following errors with `env_file` key already defined:
```
{ cmd = "...", env_file = ".env", env_file.override = true}
```

Reply:
Not overriding:
```toml
{ cmd = "...", env_file = ".env"}
```
Overriding:
```toml
{ cmd = "...", env_file.override = ".env"}
```

Reply:
You are stuck in with a boolean `override`, haha. The above however uses different structures to decide whether to override.

Reply:
I see, I've updated.